design/Exceptions.md

```diff
@@ -0,0 +1,75 @@
+The behavior of every error case in WebGPU is defined by the spec on a
+case-by-case basis. A given error has one of these behaviors:
+
+* Synchronously throws a JS exception.
+* Occurs asynchronously - one of:
+  * Causes a WebGPU object to become internally null. Produces an error log entry.
+  * Causes an operation to no-op. Produces an error log entry.
+* If the operation returns a Promise, it rejects (and maybe produces an error log entry).
+
+(If a "developer mode" is enabled, all validation errors are thrown
+synchronously, as exceptions. Device loss may or may not be synchronous and
+this behavior may be implementation-specific.
+Out-of-memory errors should NOT be made synchronous if the application would
+otherwise have an opportunity to recover from them.)
+
+**The guidelines below are meant to help choose the individual cases defined by
+the spec, but every case must be specced. This does not allow for
+"implementation-defined" behavior.** Note that an implementation can easily
+surface a synchronous error "as-if" it's asynchronous, but it cannot do the
```

Comment:
might be the language barrier, but "surface" verb rubs me the wrong way: if an error is discovered on the client side, it's already on the surface (hence, there is nothing to surface), so an opposite would make sense to me, something like: "can easily defer a synchronous error..."

Reply:
Clarified as "surface a synchronous error to the application"